pkg/limayaml/validate.go

```diff
@@ -372,6 +372,45 @@ func Validate(y *LimaYAML, warn bool) error {
 		}
 	}
 
+	if y.Rosetta.Enabled != nil {
+		if *y.Rosetta.Enabled && *y.VMType != VZ {
+			return fmt.Errorf("field `rosetta.enabled` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.NestedVirtualization != nil && *y.NestedVirtualization {
+		if *y.VMType != VZ {
+			return fmt.Errorf("field `nestedVirtualization` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.MountType != nil {
+		if err := validateMountType(y); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateMountType(y *LimaYAML) error {
+	validMountTypes := map[string]bool{
+		REVSSHFS: true,
+		NINEP:    true,
+		VIRTIOFS: true,
+		WSLMount: true,
+	}
+
+	if !validMountTypes[*y.MountType] {
+		return fmt.Errorf("field `mountType` %s, valid options are: %s, %s, %s, %s", *y.MountType, REVSSHFS, NINEP, VIRTIOFS, WSLMount)
```

Comment:
Please use `%q` consistently to quote embedded actual values.

Also mirror the text of other similar errors, e.g. for `VMType`.

```suggestion
		return fmt.Errorf("field `mountType` must be: %q, %q, %q, %q; got %q", REVSSHFS, NINEP, VIRTIOFS, WSLMount, *y.MountType)
```